addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1719,6 +1719,70 @@ export class OdooEditor extends EventTarget {
+            // break the table. If the selection includes a table cell but also
+            // elements that are out of a table, the whole table will be
+            // selected so its deletion can be handled separately.
+            const rows = [...closestElement(selectedTds[0], 'tr').parentElement.children].filter(child => child.nodeName === 'TR');
+            const firstRowCells = [...rows[0].children].filter(child => child.nodeName === 'TD' || child.nodeName === 'TH');
+            const areFullColumnsSelected = getRowIndex(selectedTds[0]) === 0 && getRowIndex(selectedTds[selectedTds.length - 1]) === rows.length - 1;
+            const areFullRowsSelected = getColumnIndex(selectedTds[0]) === 0 && getColumnIndex(selectedTds[selectedTds.length - 1]) === firstRowCells.length - 1;
+            if (areFullColumnsSelected || areFullRowsSelected) {
+                // If some full columns are selected, remove them.
+                if (areFullColumnsSelected) {
+                    const startIndex = getColumnIndex(selectedTds[0]);
+                    let endIndex = getColumnIndex(selectedTds[selectedTds.length - 1]);
+                    let currentIndex = startIndex;
+                    while (currentIndex <= endIndex) {
+                        this.execCommand('removeColumn', firstRowCells[currentIndex]);
+                        currentIndex++;
+                    }
+                }
+                // If some full rows are selected, remove them.
+                if (areFullRowsSelected) {
+                    const startIndex = getRowIndex(selectedTds[0]);
+                    let endIndex = getRowIndex(selectedTds[selectedTds.length - 1]);
+                    let currentIndex = startIndex;
+                    while (currentIndex <= endIndex) {
+                        this.execCommand('removeRow', rows[currentIndex]);
+                        currentIndex++;
+                    }
+                }
+            } else {
+                // If no full row or column is selected, empty the selected cells.
+                for (const td of selectedTds) {
+                    [...td.childNodes].forEach(child => child.remove());
+                    td.append(document.createElement('br'));
+                }
+            }
+            this.historyUnpauseSteps();
+            this.historyStep();
+        } else {
+            fullySelectedTables.forEach(table => table.remove());
+            if (this.editable.childElementCount) {
+                this._deleteRange(sel);
+            } else {
+                // Ensure the editable has content.
+                const p = document.createElement('p');
+                p.append(document.createElement('br'));
+                this.editable.append(p);
+                setSelection(p, 0);
+            }
+        }
+        if (this.deselectTable() && hasValidSelection(this.editable)) {
+            this.document.getSelection().collapseToStart();
+        }
```

Comment:
I don't really like to have `deleteRange` be about tables and the *true* `deleteRange` be relagated to `_deleteRange` though I understand the need for it with the current changes. I'm not going to argue for having `_deleteRange` be inlined in `deleteRange` as the resulting function would be incredibly long and complex. However, taking inspiration from `deselectTable`, what do you think about these two suggestions:
```js
deleteRange(sel) {
    if (this.deleteTableRange()) {
        // Code of _deleteRange;
    }
}
deleteTableRange(sel) {
    // Code of deleteRange.
    // return true or false (I don't know which one is more semantically correct) in cases where deleteRange still needs be called (and adapt the condition in deleteRange accordingly.)
}
```
OR
```
deleteRange(sel) {
    this.deleteTableRange();
    // Determine if deleteRange is still needed, maybe my naive attempt here is not enough.
    if (closestElement(sel.anchorNode, table) === closestElement(sel.focusNode, table)) {
        // Code of _deleteRange;
    }
}
deleteTableRange(sel) {
    // Code of deleteRange.
}
```
Maybe `deleteTableCellsRange` would be a better name than `deleteTableRange` ? What do you think ? Do you have better ideas ?

Reply:
Sounds good. I prefer deleteTableRange.

Reply:
Done in a separate commit.